### PR TITLE
Remove the stream lag metric.

### PR DESCRIPTION
**Pull request by cstyan**
Removes the stream lag metric since it's not possible to get a correct value and also not cause issues with metrics registration or collection until we have the WAL

Signed-off-by: Callum Styan <callumstyan@gmail.com>

### CODE REVIEW

File: clients/cmd/promtail/main.go [hunk 1]
```diff
@@ -111,7 +111,11 @@ func main() {
 		}
 	}
 
-	clientMetrics := client.NewMetrics(prometheus.DefaultRegisterer, config.Config.Options.StreamLagLabels)
+	clientMetrics := client.NewMetrics(prometheus.DefaultRegisterer)
+	if config.Options.StreamLagLabels.String() != "" {
+		level.Error(util_log.Logger).Log("msg", "client config stream_lag_labels is deprecated and the associated metric has been removed", "stream_lag_labels", config.Options.StreamLagLabels.String())
```
Comment (DylanGuedes) [inline clients/cmd/promtail/main.go:116]: maybe adding the PR link will be helpful as in the description you included the reasoning why it was removed
```suggestion
		level.Error(util_log.Logger).Log("msg", "client config stream_lag_labels is deprecated and the associated metric has been removed. See https://github.com/grafana/loki/pull/6324.", "stream_lag_labels", config.Options.StreamLagLabels.String())
```

**Review by DylanGuedes — APPROVED** [review]
lgtm, just a nit

File: clients/cmd/promtail/main.go (hunk 1 above)
Comment (cstyan) [reply clients/cmd/promtail/main.go:116]: I think this is something we would include in changelogs and release notes rather than an error message

Comment (DylanGuedes) [reply clients/cmd/promtail/main.go:116]: good point

**Review by MichelHollands — APPROVED** [review]
A rebase is needed but otherwise LGTM

**Comment by jeschkies** [comment]
@cstyan could you merge main so this can land?

**Pull request merged**